internal/storage/udev_monitor.go

```diff
@@ -0,0 +1,100 @@
+package storage
+
+import (
+	"context"
+	"log"
+	"time"
+
+	udev "github.com/farjump/go-libudev"
+	"github.com/pkg/errors"
+)
+
+type DeviceStatus string
+
+const (
+	DeviceStatusActive   DeviceStatus = "active"
+	DeviceStatusInactive DeviceStatus = "inactive"
+	DeviceStatusUnknown  DeviceStatus = "unknown"
+	udevActionAdd                     = "add"
+	udevActionRemove                  = "remove"
+)
+
+type DeviceNum struct {
+	Major int
+	Minor int
+}
+
+type UdevDevice struct {
+	DeviceNode   string
+	DeviceStatus DeviceStatus
+}
+
+type UdevMonitor struct {
+	cancel  context.CancelFunc
+	ctx     context.Context
+	monitor *udev.Monitor
+	devices map[DeviceNum]UdevDevice
+}
+
+func NewUdevMonitor(ctx context.Context, cancel context.CancelFunc) (monitor *UdevMonitor) {
+	u := udev.Udev{}
+	m := u.NewMonitorFromNetlink("udev")
+	m.FilterAddMatchSubsystemDevtype("block", "disk")
+	ret := &UdevMonitor{
+		cancel:  cancel,
+		ctx:     ctx,
+		monitor: m,
+		devices: make(map[DeviceNum]UdevDevice),
+	}
+
+	return ret
+}
+
+func (m *UdevMonitor) Start() {
+	ch, _ := m.monitor.DeviceChan(m.ctx)
+	for d := range ch {
+		device := UdevDevice{
+			DeviceNode:   d.Devnode(),
+			DeviceStatus: DeviceStatusUnknown,
+		}
+		switch d.Action() {
+		case udevActionAdd:
+			device.DeviceStatus = DeviceStatusActive
+		case udevActionRemove:
```

Comment:
Do we need to keep devices that have been removed, in memory or should we just remove them using:

```go
del m.devices[devNum]
```

?